packages/sdk/src/types/runtime/behaviors/action.ts

```diff
@@ -24,13 +29,25 @@ export class DiscreteAction {
 	private handlers: ActionHandlers = {};
 	private activeUserIds: string[] = [];
 
+	/** @hidden */
+	constructor(private name: string) {
+	}
+
 	/**
 	 * Add a handler for the given action state for when it is triggered.
 	 * @param actionState The action state that the handle should be assigned to.
 	 * @param handler The handler to call when the action state is triggered.
 	 */
-	public on(actionState: ActionState, handler: ActionHandler): this {
+	public on(
+		context: Context,
+		actorId: string,

```

Comment:
These arguments seem very awkward and out of place here.  Can't we simply pass in a reference to the behavior that owns this action, and thus has access to the context and actor ID from there?  Seems cleaner.

Reply:
I guess we have to live with this here.  Not the worst thing in the world.  Maybe a better way to refactor this in the future.

Reply:
Glad you're ok with it, at least temporarily. It's all internal thankfully. I think we should mark this method @hidden.

---
In reply to: [316876865](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316876865) [](ancestors = 316876865)